### PR TITLE
Fix DotEnv class turning `export` to empty string

### LINKED ISSUE #6621: Bug: Unable to connect to the database when name contains word `export`
### PHP Version

8.1

### CodeIgniter4 Version

4.2.1

### CodeIgniter4 Installation Method

Composer (using `codeigniter4/appstarter`)

### Which operating systems have you tested for this bug?

Windows

### Which server did you use?

cli

### Database

PostgreSQL

### What happened?

Unable to connect to the database when database group name contains word _export_, even is it's only a partial name, ie. _exports_data_ will also fail.

### Steps to Reproduce

Define db group with word `export` in .env file:
```
database.exports_test.hostname = localhost
database.exports_test.database = dev_db
database.exports_test.username = postgres
database.exports_test.password = postgres
database.exports_test.DBDriver = Postgre
database.exports_test.DBDebug = FALSE
database.exports_test.schema = exports_test
database.exports_test.port = 5432
```

define this group in app/Config/Database.php:
```
public $exports_test= [
    'hostname' => '',
    'schema' => '',
    'username' => '',
    'password' => '',
    'database' => '',
    'DBDriver' => '',
    'DBPrefix' => '',
    'pConnect' => false,
    'DBDebug'  => (ENVIRONMENT !== 'production'),
    'cacheOn'  => false,
    'cacheDir' => '',
    'charset'  => 'utf8',
    'DBCollat' => 'utf8_general_ci',
    'swapPre'  => '',
    'encrypt'  => false,
    'compress' => false,
    'strictOn' => false,
    'failover' => [],
];
```

try to run a query against the db. You will receive `You have not selected a database type to connect to` error message

### Expected Output

Query results

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by paulbalandan**
**Description**
Fixes #6621 

I think the bug is on the incorrect sanitization of exported env variables with the syntax `export varname=value`.

**Checklist:**
- [x] Securely signed commits
- [ ] Component(s) with PHPDoc blocks, only if necessary or adds value
- [x] Unit testing, with >80% coverage
- [ ] User guide updated
- [x] Conforms to style guide

### CODE REVIEW

File: system/Config/DotEnv.php [hunk 1]
```diff
@@ -116,7 +116,8 @@ public function normaliseVariable(string $name, string $value = ''): array
         $value = trim($value);
 
         // Sanitize the name
-        $name = str_replace(['export', '\'', '"'], '', $name);
```
Comment (kenjis) [inline system/Config/DotEnv.php:119]: What is the purpose of deleting these strings?

**Comment by kenjis** [comment]
Is `export varname=value` valid syntax for `.env`?
It seems no such explanation in the user guide.

File: system/Config/DotEnv.php (hunk 1 above)
Comment (paulbalandan) [reply system/Config/DotEnv.php:119]: I'm not entirely sure, but I think this was meant to make `'foo'` or `"foo"` to just `foo`

**Comment by paulbalandan** [comment]
> Is `export varname=value` valid syntax for `.env`? It seems no such explanation in the user guide.

I think that was the intention.

**Comment by paulbalandan** [comment]
Okay. Since there was mention that the code comes mostly from `vlucas/phpdotenv`, I checked the repo and found this code in v2.0.0
https://github.com/vlucas/phpdotenv/blob/058b9b03b945b6c3eced241935c45822fd2bf4e8/src/Loader.php#L249-L254

It seems our DotEnv class is based on an old code.

File: tests/system/Config/fixtures/.env [hunk 2]
```diff
@@ -4,6 +4,11 @@ SPACED="with spaces"
 
 NULL=
 
+char.expo.foo="exported foo"
+character.export.var=variable
+export char.var=character
+export char.exports=imports
+
```
Comment (MGatner) [inline tests/system/Config/fixtures/.env:11]: Can you add a test for this spacing?
```suggestion
fruit.export = "banana"
```

**Review by MGatner — CHANGES_REQUESTED** [review]
Looks good, an edge case I would like to see covered.

**Comment by paulbalandan** [comment]
> Just saw your comment - should we use an actual package for this, either via Composer or unlined like we do with Kint? Seems like a high-risk security area to be based off old third-party code.

Well I'm in favor. Less maintenance burden on us.

**Comment by kenjis** [comment]
What security concern is in DotEnv?
It seems a few methods were borrowed from vlucas/phpdotenv when we see the code comment.

Config is a part that has slowed down considerably with CI4, and I don't want to see it get any heavier.

**Comment by kenjis** [comment]
It seems vlucas/phpdotenv support `export varname=value`:
https://github.com/vlucas/phpdotenv/blob/master/tests/fixtures/env/exported.env

But for what?

**Review by MGatner — APPROVED** [review]
I'm good with this fix for now. I do agree that Config has gotten rather bloated, but I think DotEnv is distinct enough (and necessary) that I would be in favor of some research at least about outsourcing it.

**Comment by kenjis** [comment]
This is not related to this PR, but I've found this PHP extention:
https://github.com/sjinks/php-ext-dotenv
> The parser is compatible with that of [Node.js dotenv](https://www.npmjs.com/package/dotenv) version 10.0 (test cases use the same environment file to test parsing).

File: tests/system/Config/fixtures/.env [hunk 3]
```diff
@@ -4,6 +4,12 @@ SPACED="with spaces"
 
 NULL=
 
+char.expo.foo="exported foo"
+character.export.var=variable
+export char.var=character
+export char.exports=imports
+fruit.export = "banana"
```
Comment (kenjis) [inline tests/system/Config/fixtures/.env:11]: Before: $_ENV
```
    'char.expo.foo' => string (12) "exported foo"
    'character..var' => string (8) "variable"
    ' char.var' => string (9) "character"
    ' char.s' => string (7) "imports"
    'fruit.' => string (6) "banana"
```
After: $_ENV
```
    'char.expo.foo' => string (12) "exported foo"
    'character.export.var' => string (8) "variable"
    'char.var' => string (9) "character"
    'char.exports' => string (7) "imports"
    'fruit.export' => string (6) "banana"
```

Comment (kenjis) [reply tests/system/Config/fixtures/.env:11]: Node.js dotenv@16.0.3:
```
  'char.expo.foo': 'exported foo',
  'character.export.var': 'variable',
  'char.var': 'character',
  'char.exports': 'imports',
  'fruit.export': 'banana'
```
It seems that `export` is handled the same way in Node dotenv as in this PR.

**Review by kenjis — APPROVED** [review]
As the bug fix, this has no problem.

**Comment by paulbalandan** [comment]
Thanks for the reviews
